spatial-tracking-explainer.md

```diff
@@ -0,0 +1,382 @@
+
+There are three types of frames of reference: `bounded`, `unbounded`, and `stationary`.  A bounded experience is one in which the user will move around their physical environment to fully interact, but will not need to travel beyond a fixed boundary defined by the XR hardware.  An unbounded experience is one in which a user is able to freely move around their physical environment and travel significant distances.  A stationary experience is one which does not require the user to move around in space, and includes "seated" or "standing" experiences.
+
+The critical aspect of this approach is that the User Agent (or underlying platform) is responsible for providing consistently behaved lower-capability `XRFrameOfReference` objects even when running on a higher-capability tracking system.  That said, not all experiences will work on all XR hardware and not all XR hardware will support all experiences (see Appendix A: XRFrameOfReference Availability).  In the spirit of [progressive enhanncement](https://developer.mozilla.org/en-US/docs/Glossary/Progressive_Enhancement), it is strongly recommended that developers select the least capable `XRFrameOfReference` that suffices for the experience they are building.  Requesting a more capable frame of reference will artificially restrict the set of XR devices their experience will otherwise be viewable from.
+
+### Bounded Frame of Reference
+A _bounded_ experience is one in which a user moves around their physical environment to fully interact, but will not need to travel beyond a pre-established boundary.  A _bounded_ experience is similar to a _unbounded_ experience in that both rely on XR hardware capable of tracking a users' locomotion.  However, _bounded_ experiences are explicitly focused on nearby content which allows them to target XR hardware that requires a preconfigured play area as well as XR hardware able to track location freely.
+
+The origin of this type will be initialized at a position on the floor for which a boundary can be provided to the app, defining an empty region where it is safe for the user to move around. The y value will be 0 at floor level, while the exact x, z, and orientation values will be initialized based on the conventions of the underlying platform for room-scale experiences. Platforms where the user defines a fixed room-scale origin and boundary may initialize the remaining values to match the room-scale origin. Users with fixed-origin systems are familiar with this behavior, however developers may choose to be extra resilient to this situation by building UI to guide users back to the origin if they are too far away. Platforms that generally allow for unbounded movement may display UI to the user during the asynchronous request, asking them to define or confirm such a floor-level boundary near the user's current location.
```

Comment:
How are we dealing with non-flat floors?  It seems reasonable to assume (perhaps) that bounded frames have a flat floor;  are we going to assume (perhaps telling users) that if they defined a bounded frame for an unbounded device, it should have a flat floor?  

It's probably "fine" to assume this;  the failure case is that users try to use a bounded experience somewhere where the floor isn't flat, and they can't reach some content (because it's too high, or it's under the real floor).

Reply:
Yes, that's actually one of the simplifying assumptions developers will be able to make; `XRBoundedFrameOfReference` objects are expected to have a flat floor within the bounds.